README.md
# Leapfrog-Firmware
Firmware based on Marlin for Leapfrog 3D Printers

Started out as a fork from Marlin 1.0.0 [Reference](https://github.com/MarlinFirmware/Marlin) or even earlier 

## Changes
- Heavily slimmed down and removed all bloat(LCD, SD, COREXY, DELTA)
- Many Leapfrog specific features and adjustments


CHANGELOG.md
# Changelog

Changes to Leapfrog-Firmware

## 3.0.0

Set Y-shift fix in accordance with new LMC wiring

## 2.8

- Added detection and support for high temperature hotend
- Support for PT-100 temperature sensor
- Support for Z-max endstop

###Config changes
| Name | Old | New | Comment
|---|---|---|---
TEMP_SENSOR_0 | 1 | -3 | PT-100 sensor
TEMP_SENSOR_1 | 1 | -3 | PT-100 sensor


## 2.7

- Changed bed coordinates for X
- Changed parking distance 
- Changed Z-homing speed 
- Added min_bed_temp warning and check
- Added ability to skip temperature sync when enabling sync mode
- Added ability to interrupt M140 and M190 commands with a M108 command
- Removed individual parking positions
- Fixed parking position and calibration bug

###Config changes
| Name | Old | New
|---|---|---
HOMING_FEEDRATE | `100*60, 100*60, 20*60, 0` | `100*60, 100*60, 600, 0`
DEFAULT_PARK_OFFSET | `10.0` | `1.0`

| Buildvolume | Old | New 
|---| --- | --- |
| Xmin | `0.0` | **`-37.0`** 
| Xmax | `367.0` | **`330.0`**
| Ymin | `-33.0` | `-33.0` 
| Ymax| `322.0` | `322.0` 
| Zmin| `0.0` | `0.0` 
| Zmax | `205.0`| `205.0`


## 2.6

- Changed bed volume and homing position to make sure that actual buildvolume is ok
- Fixed critical bug on software endstop declaration

| Buildvolume | Old | New 
|---| --- | --- |
| Xmin | `0.0` | `0.0`
| Xmax | `367.0` | `367.0`
| Ymin | `0.0` | **`-33.0`** 
| Ymax| `355.0` | **`322.0`** 
| Zmin| `0.0` | `0.0` 
| Zmax | `205.0`| `205.0`


## 2.5

- Fixed temperature reporting. 2.4 had an output error
- Changed max feedrate of Z-axis due to steps max
- Changed bed width calibration to negative output in M115

###Config changes
| Name | Old | New
|---|---|---
DEFAULT_MAX_FEEDRATE | `400, 400, 40, 200` | `400, 400, 20, 200`

## 2.4

- Changed temperature reporting
Removed the active extruder temperature.
OLD: `T: 20 / 0 B: 20 / 0 T0: 20 / 0 T1: 21 /0 @: 0`
NEW: `T0: 20 / 0 T1: 21 /0 B: 20 / 0  @: 0`
This is done because Mirror mode makes "active extruder" always be T1 that makes LUI act weird.

###Config changes
| Name | Old | New
|---|---|---
| DEFAULT_MAX_ACCELERATION | `450, 450, 100, 30` | `400,400,100,30`
| DEFAULT_ACCELERATION | `1000` | `400`
| DEFAULT_XYJERK | `25` | `20`


## Beta

- Dual X Calibration M219
- RTD sensor support
- No movements for M605
- G28 is now allowed (and should be required) after M605
- Fixed toolchange during sync mode
- Updated M115 with bed width calibration
- Updated acceleration/velocity settings Setting 

###Config changes
| Name | Old | New
|---|---|---
DEFAULT_MAX_FEEDRATE | `400, 400, 40, 400` | `400, 400, 40, 200`
DEFAULT_MAX_ACCELERATION | `1250,1250,100,10000` | `450,450,100,30` 
DEFAULT_RETRACT_ACCELERATION | `5000` | `1250`
DEFAULT_XYJERK | `15.0` | `25.0`

pt100tables.h
#ifndef PT100TABLES_H_
#define PT100TABLES_H_

#include "Marlin.h"

#if (defined(HEATER_0_USES_DETECTION) || defined(HEATER_1_USES_DETECTION))
// PT-100 temperature sensor.
// PT-100 is positive temperature coefficient: no inversion applied to the raw measured ADC value
// Measured ADC result is 10 bits, put into an unsigned 16 bit integer (rght adjusted).
// 0V corresponds to 0, 5V corresponds to 1023
// TODO provide documentation and link
const short temptable_pt100[][2] PROGMEM = { 
{ OVERSAMPLENR * 400, 0 },
{ OVERSAMPLENR * 407, 5 },
{ OVERSAMPLENR * 415, 10 },
{ OVERSAMPLENR * 423, 15 },
{ OVERSAMPLENR * 430, 20 },
{ OVERSAMPLENR * 438, 25 },
{ OVERSAMPLENR * 445, 30 },
{ OVERSAMPLENR * 453, 35 },
{ OVERSAMPLENR * 460, 40 },
{ OVERSAMPLENR * 468, 45 },
{ OVERSAMPLENR * 475, 50 },
{ OVERSAMPLENR * 483, 55 },
{ OVERSAMPLENR * 490, 60 },
{ OVERSAMPLENR * 498, 65 },
{ OVERSAMPLENR * 505, 70 },
{ OVERSAMPLENR * 512, 75 },
{ OVERSAMPLENR * 520, 80 },
{ OVERSAMPLENR * 527, 85 },
{ OVERSAMPLENR * 534, 90 },
{ OVERSAMPLENR * 542, 95 },
{ OVERSAMPLENR * 549, 100 },
{ OVERSAMPLENR * 556, 105 },
{ OVERSAMPLENR * 564, 110 },
{ OVERSAMPLENR * 571, 115 },
{ OVERSAMPLENR * 578, 120 },
{ OVERSAMPLENR * 585, 125 },
{ OVERSAMPLENR * 593, 130 },
{ OVERSAMPLENR * 600, 135 },
{ OVERSAMPLENR * 607, 140 },
{ OVERSAMPLENR * 614, 145 },
{ OVERSAMPLENR * 621, 150 },
{ OVERSAMPLENR * 628, 155 },
{ OVERSAMPLENR * 635, 160 },
{ OVERSAMPLENR * 643, 165 },
{ OVERSAMPLENR * 650, 170 },
{ OVERSAMPLENR * 657, 175 },
{ OVERSAMPLENR * 664, 180 },
{ OVERSAMPLENR * 671, 185 },
{ OVERSAMPLENR * 678, 190 },
{ OVERSAMPLENR * 685, 195 },
{ OVERSAMPLENR * 692, 200 },
{ OVERSAMPLENR * 699, 205 },
{ OVERSAMPLENR * 706, 210 },
{ OVERSAMPLENR * 713, 215 },
{ OVERSAMPLENR * 720, 220 },
{ OVERSAMPLENR * 726, 225 },
{ OVERSAMPLENR * 733, 230 },
{ OVERSAMPLENR * 740, 235 },
{ OVERSAMPLENR * 747, 240 },
{ OVERSAMPLENR * 754, 245 },
{ OVERSAMPLENR * 761, 250 },
{ OVERSAMPLENR * 767, 255 },
{ OVERSAMPLENR * 774, 260 },
{ OVERSAMPLENR * 781, 265 },
{ OVERSAMPLENR * 788, 270 },
{ OVERSAMPLENR * 794, 275 },
{ OVERSAMPLENR * 801, 280 },
{ OVERSAMPLENR * 808, 285 },
{ OVERSAMPLENR * 815, 290 },
{ OVERSAMPLENR * 821, 295 },
{ OVERSAMPLENR * 828, 300 },
{ OVERSAMPLENR * 835, 305 },
{ OVERSAMPLENR * 841, 310 },
{ OVERSAMPLENR * 848, 315 },
{ OVERSAMPLENR * 854, 320 },
{ OVERSAMPLENR * 861, 325 },
{ OVERSAMPLENR * 868, 330 }, 
{ OVERSAMPLENR * 874, 335 },
{ OVERSAMPLENR * 881, 340 },
{ OVERSAMPLENR * 887, 345 },
{ OVERSAMPLENR * 894, 350 },
{ OVERSAMPLENR * 900, 355 },
{ OVERSAMPLENR * 907, 360 },
{ OVERSAMPLENR * 913, 365 },
{ OVERSAMPLENR * 920, 370 },
{ OVERSAMPLENR * 926, 375 },
{ OVERSAMPLENR * 932, 380 },
{ OVERSAMPLENR * 939, 385 },
{ OVERSAMPLENR * 945, 390 },
{ OVERSAMPLENR * 951, 395 },
{ OVERSAMPLENR * 958, 400 },
//Extra high temps
{ OVERSAMPLENR * 964, 405 },
{ OVERSAMPLENR * 971, 410 },
{ OVERSAMPLENR * 977, 415 },
{ OVERSAMPLENR * 984, 420 },
{ OVERSAMPLENR * 990, 425 },
{ OVERSAMPLENR * 997, 430 },
{ OVERSAMPLENR * 1003, 435 }
};
#define temptable_pt100_len (sizeof(temptable_pt100) / sizeof(temptable_pt100[0]))
#endif

#endif /* PT100TABLES_H_ */
